litgpt/scripts/merge_lora.py

```diff
@@ -43,16 +43,16 @@ def merge_lora(
     fabric = L.Fabric(devices=1, precision=precision, accelerator="cpu")
     config = Config.from_file(checkpoint_dir / "model_config.yaml", **lora_params)
 
-    with fabric.init_module(empty_init=True):
+    with fabric.init_module(), torch.device("meta"):
         model = GPT(config)
 
     lora_path = checkpoint_dir / "lit_model.pth.lora"
-    pretrained_checkpoint = lazy_load(pretrained_checkpoint_dir / "lit_model.pth")
-    lora_checkpoint = lazy_load(lora_path)
+    pretrained_checkpoint = torch.load(str(pretrained_checkpoint_dir / "lit_model.pth"), mmap=True)
+    lora_checkpoint = torch.load(str(lora_path), mmap=True)
```

Comment:
Sorry for the dumb question, but how does this help?
Surely I'm missing something, but I thought that these changes are beneficial when we want to materialize weights on a GPU without an intermediate CPU RAM step.

Reply:
`lazy_load` and `torch.load(mmap)` are basically the same. But the latter supports `assign=True` and the former doesnt.

The assign change below is what should improve the memory usage

Reply:
I'm so dumb, it's unbelievable 🙉. Let's blame Monday for it.
Instead of storing a model with random weights and pretrained+LoRA weights at the same time, we initialize a model on a meta device and then materialize it with the correct weights.